scripts/mh_windowfill.pl

```diff
@@ -0,0 +1,244 @@
+#
+##############################################################################
+#
+# fill windows so scrolling starts bottom-up instead of top-down
+#
+# known issues:
+# 	- /CLEAR will reset to top-down
+#	- it is possible to confuse the script into not filling with a combination
+#	  of script load/unloads and window resizes. but it requires effort
+#
+# history:
+#	v1.00 (20151116)
+#		initial release
+#
+
+use v5.14.2;
+
+use strict;
+
+##############################################################################
+#
+# irssi head
+#
+##############################################################################
+
+use Irssi 20100403;
+use Irssi::TextUI;
+
+our $VERSION = '1.00';
+our %IRSSI   =
+(
+	'name'        => 'mh_windowfill',
+	'description' => 'fill windows so scrolling starts bottom-up instead of top-down',
+	'license'     => 'BSD',
+	'authors'     => 'Michael Hansen',
+	'contact'     => 'mh on IRCnet #help',
+	'url'         => 'irc://open.ircnet.net',
+);
+
+##############################################################################
+#
+# global variables
+#
+##############################################################################
+
+our $windowfill_running = 0;
+
+##############################################################################
+#
+# common support functions
+#
+##############################################################################
+
+sub trim_spacelike($)
+{
+   my ($string) = @_;
+
+   if (defined($string))
+   {
+      $string =~ s/^\s+//g;
+      $string =~ s/\s+$//g;
+
+   } else {
+
+      $string = '';
+   }
+
+   return($string);
+}
+
+##############################################################################
+#
+# script functions
+#
+##############################################################################
+
+sub windowfill($)
+{
+	my ($window) = @_;
+
+	if (ref($window) ne 'Irssi::UI::Window')
+	{
+		die();
+	}
+
+	#
+	# fill window with empty lines and move already printed lines to the bottom
+	#
+	if (($window->view()->{'ypos'} + 2) <= $window->{'height'})
+	{
+
+		while (($window->view()->{'ypos'} + 2) <= $window->{'height'})
+		{
+			$window->print('', MSGLEVEL_CLIENTCRAP | MSGLEVEL_NEVER | MSGLEVEL_NO_ACT | MSGLEVEL_NOHILIGHT);
+		}
+
+		my $linecount  = $window->{'height'};
+		my $line       = $window->view()->get_lines();
+
+		while ((ref($line) eq 'Irssi::TextUI::Line') and $linecount)
+		{
+			my $linetext = $line->get_text(1);
+
+			if ($linetext ne '')
+			{
+				# reprint line
+				$window->print($linetext, MSGLEVEL_CLIENTCRAP | MSGLEVEL_NEVER | MSGLEVEL_NO_ACT | MSGLEVEL_NOHILIGHT);
+				$line = $line->next();
+				$window->view()->remove_line($line->prev());
+
+			} else {
+
+				# skip empty line
+				$line = $line->next();
+			}
+
+			$linecount--;
+		}
+	}
+
+	return(1);
+}
+
+sub windowfill_all()
+{
+	#
+	# fill all windows with empty lines
+	#
+	for my $window (Irssi::windows())
+	{
+		if (ref($window) ne 'Irssi::UI::Window')
```

Comment:
Again, this also protects me while i develop the script, it is not just about the offchance that irssi borks and calls something without the right arguments.

and i didnt know i could do that... however, it will not be without most of what you call useless code, the script is not live-tested enough for that yet.